url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/imex/FisExporter.java

```diff
@@ -0,0 +1,527 @@
+            inputVariables.add(inputVariable);
+        }
+        for (OutputVariable outputVariable : engine.getOutputVariables()) {
+            outputVariables.add(outputVariable);
+        }
+
+        StringBuilder result = new StringBuilder();
+        result.append(translate(propositions, inputVariables)).append(", ");
+        result.append(translate(rule.getConsequent().getConclusions(), outputVariables));
+        result.append(String.format("(%s)", str(rule.getWeight())));
+        String connector;
+        if (operators.isEmpty()) {
+            connector = "1";
+        } else if (Rule.FL_AND.equals(operators.get(0).getName())) {
+            connector = "1";
+        } else if (Rule.FL_OR.equals(operators.get(0).getName())) {
+            connector = "2";
+        } else {
+            connector = operators.get(0).getName();
+        }
+        result.append(" : ").append(connector);
+        return result.toString();
+    }
+
+    protected String extractType(Engine engine) {
+        if (engine.type() == Engine.Type.NONE) {
+            return "none";
+        } else if (engine.type() == Engine.Type.MAMDANI || engine.type() == Engine.Type.LARSEN) {
+            return "mamdani";
+        } else if (engine.type() == Engine.Type.TAKAGI_SUGENO) {
+            return "sugeno";
+        } else if (engine.type() == Engine.Type.TSUKAMOTO) {
+            return "tsukamoto";
+        } else if (engine.type() == Engine.Type.INVERSE_TSUKAMOTO) {
+            return "inverse tsukamoto";
+        }
+        return "unknown";
+    }
+
+    protected String extractAccumulation(Engine engine) {
+        SNorm accumulation = null;
+        for (OutputVariable outputVariable : engine.getOutputVariables()) {
+            SNorm other = outputVariable.fuzzyOutput().getAccumulation();
+            if (accumulation == null) {
+                accumulation = other;
+            } else if (!accumulation.getClass().equals(other.getClass())) {
+                throw new RuntimeException("[export error] "
+                        + "fis files require all output variables to have the "
+                        + "same accumulation operator");
+            }
+        }
+        return toString(accumulation);
+    }
+
+    protected String extractDefuzzifier(Engine engine) {
+        Defuzzifier defuzzifier = null;
+        for (OutputVariable outputVariable : engine.getOutputVariables()) {
+            Defuzzifier other = outputVariable.getDefuzzifier();
+            if (defuzzifier == null) {
+                defuzzifier = other;
+            } else if (!defuzzifier.getClass().equals(other.getClass())) {
+                throw new RuntimeException("[export error] "
+                        + "fis files require all output variables "
+                        + "to have the same defuzzifier");
+            }
+        }
+        return toString(defuzzifier);
+    }
+
+    protected String translate(List<Proposition> propositions, List<Variable> variables) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.